crates/interledger-stream/src/packet.rs

```diff
@@ -173,13 +173,26 @@ impl StreamPacket {
         let num_frames = reader.read_var_uint()?;
         let frames_offset = buffer_unencrypted.len() - reader.len();
 
-        // Try reading through all the frames to make sure they can be parsed correctly
+        let mut reader = &buffer_unencrypted[frames_offset..];
+        for _ in 0..num_frames {
+            // First byte is the frame type
+            reader.skip(1)?;
+            reader.skip_var_octet_string()?;
+        }
+
+        let junk_data_len = reader.len();
+
+        if junk_data_len > 0 {
+            let _ = buffer_unencrypted.split_off(buffer_unencrypted.len() - junk_data_len);
```

Comment:
could add a comment here with a link to the stream rfc about the junk bytes. Something like:

```suggestion
            // trailing bytes are supported for future compatibility, see
            // https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md#52-stream-packet
            let _ = buffer_unencrypted.split_off(buffer_unencrypted.len() - junk_data_len);
```

Also this could just be `buffer_unencrypted.split_off(...)`; there's no `#[must_use]` on the `split_off`.

Reply:
Though now I wonder if we should keep the trailer either way, just explicitly support it via the builder?

The rfc looks fuzzy on this as in it doesn't say that:
- successfully decrypted stream packet
- with ignored junk bytes (hopefully padded with leading zeroes)
- must be serialized again with the not understood trailing junk

However perhaps the "successfully decrypted stream packet" is a give-away that this will never be serialized again, with #704 fixed.

Perhaps supporting it through the builder is something that can be done later on if needed.